File: Python/Dosen.py
from Civitas import Civitas
from Prodi import Prodi

class Dosen(Civitas):
    # dosen adalah child dari civitas
    
    # atribut private
    __nip = ""
    __pendAkhir = ""
    __keahlian = ""
    __prodiA = Prodi()
    
    #constructor
    def __init__(self, nik="", name="", gender="", univ="", email="", fakultas = "", nip = "", keahlian = "", pendAkhir ="", prodiA = Prodi()):
        super().__init__(nik, name, gender, univ, email, fakultas)
        self.__nip = nip
        self.__keahlian = keahlian
        self.__pendAkhir = pendAkhir
        self.__prodiA = prodiA
    
    # getter setter
    def setNip(self, nip):
        self.__nip = nip
    
    def getNip(self):
        return self.__nip
    
    def setPend(self, pendAkhir):
        self.__pendAkhir = pendAkhir
    
    def getPend(self):
        return self.__pendAkhir

    def setAhli(self, keahlian):
        self.__keahlian = keahlian
    
    def getAhli(self):
        return self.__keahlian
    
    def setProdi(self, prodiA):
        self.__prodiA = prodiA
    
    def getProdi(self):
        return self.__prodiA

File: CPP/Program/Dosen.cpp
#include "header.hh"

// Constructor
Dosen::Dosen() : Civitas()
{
    nip = "";
    penAkhir = "";
    keahlian = "";
}

// Constructor dengan atribut
Dosen::Dosen(string nik, string name, string gender, string univ, string email, string fakultas, string nip, string penAkhir, string keahlian, Prodi prodiC, Course courseD) : Civitas(nik, name, gender, univ, email, fakultas)
{
    this->nip = nip;
    this->penAkhir = penAkhir;
    this->keahlian = keahlian;
    this->prodiC = prodiC;
}

// Getter Setter
string Dosen::getNIP()
{
    return nip;
}

void Dosen::setNIP(string nip)
{
    this->nip = nip;
}

string Dosen::getAhli()
{
    return keahlian;
}

void Dosen::setAhli(string keahlian)
{
    this->keahlian = keahlian;
}

string Dosen::getAkhir()
{
    return penAkhir;
}

void Dosen::setAkhir(string penAkhir)
{
    this->penAkhir = penAkhir;
}

Prodi Dosen::getProdiC()
{
    return prodiC;
}

void Dosen::setProdiC(Prodi prodiC)
{
    this->prodiC = prodiC;
}

list<Course> Dosen::getCourseD()
{
    return courseD;
}

void Dosen::setCourseD(Course courseD)
{
    this->courseD.push_back(courseD);
}

// Destructor
Dosen::~Dosen() {}